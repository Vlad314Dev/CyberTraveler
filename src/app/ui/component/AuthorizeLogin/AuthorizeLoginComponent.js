import './AuthorizeLoginStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

class AuthorizeLoginComponent extends PureComponent
{
    static propTypes = {
        handleSubmit: PropTypes.func.isRequired
    };

    render()
    {
        const { 
            handleSubmit
        } = this.props;

        return (
            <div block="AuthorizeSignup">
                <form onSubmit={ handleSubmit } block="Form">
                    <div block="Form" elem="Field">
                        <label htmlFor="nickname">Nickname:</label>
                        <input type="text" name="nickname" id="nickname" />
                    </div>
                    <div block="Form" elem="Field">
                        <label htmlFor="password">Password:</label>
                        <input type="password" name="password" id="password" />
                    </div>
                    <div block="Form" elem="Submit">
                        <button type="submit">Login</button>
                    </div>
                </form>
            </div>
        );
    }
}

export default AuthorizeLoginComponent;
