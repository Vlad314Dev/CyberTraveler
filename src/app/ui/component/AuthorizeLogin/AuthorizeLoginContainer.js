import React, { PureComponent } from 'react';
import { connect } from 'react-redux';

import LeaderboardComponent from './AuthorizeLoginComponent';

class AuthorizeLoginContainer extends PureComponent
{
    containerFunctions = {
        handleSubmit: this.handleSubmit.bind(this)
    };

    handleSubmit(data)
    {
        console.log(this, data);
    }

    render()
    {
        return (
            <LeaderboardComponent 
                { ...this.props }
                { ...this.state }
                { ...this.containerFunctions }
            />
        );
    }
}

export default connect(null, null)(AuthorizeLoginContainer);
