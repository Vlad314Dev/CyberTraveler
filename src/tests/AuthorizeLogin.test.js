/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';
import configureMockStore from 'redux-mock-store';

import AuthorizeLogin from '../app/ui/component/AuthorizeLogin';
import AuthorizeLoginComponent from '../app/ui/component/AuthorizeLogin/AuthorizeLoginComponent';

const mockStore = configureMockStore();
const store = mockStore({});

describe('AuthorizeLogin', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <AuthorizeLogin 
                store={ store }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <AuthorizeLogin 
                store={ store }
            />
        ).dive().find('AuthorizeLoginComponent').dive();

        expect(component.getElements()).toMatchSnapshot();
    });

    it('Should update component state on nickname change', () => {
        const container = shallow(
            <AuthorizeLogin 
                store={ store }
            />
        ).dive();
        const component = container.find('AuthorizeLoginComponent').dive();
        const input = component.find('#nickname');
        const testValue = 'test';
        
        input.props().onChange({
            target: {
                value: testValue
            }
        });

        expect(container.state('nickname')).toEqual(testValue);
    });

    it('Should update component state on password change', () => {
        const container = shallow(
            <AuthorizeLogin 
                store={ store }
            />
        ).dive();
        const component = container.find('AuthorizeLoginComponent').dive();
        const input = component.find('#password');
        const testValue = 'test';
        
        input.props().onChange({
            target: {
                value: testValue
            }
        });

        expect(container.state('password')).toEqual(testValue);
    });

    it('Should submit form', () => {
        const testProps = {
            handleSubmit: jest.fn(),
            handleNickname: jest.fn(),
            handlePassword: jest.fn(),
            loading: false
        };
        
        const component = shallow(
            <AuthorizeLoginComponent
                store={ store }
                { ...testProps }
            />
        );
        const nicknameInput = component.find('#nickname');
        const passwordInput = component.find('#password');
        const form = component.find('form');

        nicknameInput.props().onChange({
            target: {
                value: 'test'
            }
        });
        passwordInput.props().onChange({
            target: {
                value: 'test'
            }
        });
        
        form.simulate('submit');

        expect(testProps.handleSubmit).toHaveBeenCalled();
    });
});
