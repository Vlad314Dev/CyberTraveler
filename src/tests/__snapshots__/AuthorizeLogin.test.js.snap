// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorizeLogin Should render layout 1`] = `
Array [
  <Unknown
    color="#fff"
    component="div"
    loaded={true}
    loadedClassName="loadedContent"
    parentClassName="loader"
  >
    <div
      className="AuthorizeSignup"
    >
      <form
        className="Form"
        onSubmit={[Function]}
      >
        <div
          className="Form__Field"
        >
          <label
            htmlFor="nickname"
          >
            Nickname:
          </label>
          <input
            id="nickname"
            name="nickname"
            onChange={[Function]}
            type="text"
          />
        </div>
        <div
          className="Form__Field"
        >
          <label
            htmlFor="password"
          >
            Password:
          </label>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </div>
        <div
          className="Form__Submit"
        >
          <button
            onSubmit={[Function]}
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </Unknown>,
]
`;
